### ISSUE TITLE
[ojelectronics] Binding stops working after server timeout

### ISSUE THREAD

**Issue by seime** [openhab/openhab-addons#11531]
Problem: Bridge hits an unexpected timeout response from the server, then stops working.

OpenHab version: 3.1.0

Thing configuration:

```
Bridge ojelectronics:ojcloud:account "Micromatic Bridge" @ "Hele huset" [ 
  userName="xxxxx", 
  password="xxxx", 
  apiKey="xxxx", 
  customerId="15" 
  ] {
    Thing owd5 gulvtermostatBadNede "Bad nede" [ serialNumber="xxx" ]
  ....
```
Logs:
```
2021-11-05 18:56:16.386 [TRACE] [jelectronics.internal.OJCloudHandler] - OJElectronicsCloudHandler.handleRefreshDone(org.openhab.binding.ojelectronics.internal.models.groups.GroupContentResponseModel@f88730)
2021-11-05 19:00:27.083 [TRACE] [ics.internal.services.RefreshService] - refresh <html><head><title>500 - The request timed out.</title></head><body>  <font color ="#aa0000">         <h2>500 - The request timed out.</h2></font>  The web server failed to respond within the specified time.</body></html>
2021-11-05 19:00:27.084 [DEBUG] [ics.internal.services.RefreshService] - Error mapping Result to model
com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:226) ~[bundleFile:?]
        at com.google.gson.Gson.fromJson(Gson.java:932) ~[bundleFile:?]
        at com.google.gson.Gson.fromJson(Gson.java:897) ~[bundleFile:?]
        at com.google.gson.Gson.fromJson(Gson.java:846) ~[bundleFile:?]
        at com.google.gson.Gson.fromJson(Gson.java:817) ~[bundleFile:?]
        at org.openhab.binding.ojelectronics.internal.services.RefreshService.handleRefreshDone(RefreshService.java:143) [bundleFile:?]
        at org.openhab.binding.ojelectronics.internal.services.RefreshService$1.onComplete(RefreshService.java:125) [bundleFile:?]
        at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:218) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:210) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.client.HttpReceiver.terminateResponse(HttpReceiver.java:481) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.client.HttpReceiver.terminateResponse(HttpReceiver.java:461) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.client.HttpReceiver.responseSuccess(HttpReceiver.java:424) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.messageComplete(HttpReceiverOverHTTP.java:365) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.http.HttpParser.handleContentMessage(HttpParser.java:591) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.http.HttpParser.parseContent(HttpParser.java:1716) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:1545) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.parse(HttpReceiverOverHTTP.java:204) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.process(HttpReceiverOverHTTP.java:144) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.receive(HttpReceiverOverHTTP.java:79) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.client.http.HttpChannelOverHTTP.receive(HttpChannelOverHTTP.java:131) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.client.http.HttpConnectionOverHTTP.onFillable(HttpConnectionOverHTTP.java:169) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint.onFillable(SslConnection.java:540) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.io.ssl.SslConnection.onFillable(SslConnection.java:395) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.io.ssl.SslConnection$2.succeeded(SslConnection.java:161) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:383) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:882) [bundleFile:9.4.40.v20210413]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1036) [bundleFile:9.4.40.v20210413]
        at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
        at com.google.gson.stream.JsonReader.beginObject(JsonReader.java:386) ~[bundleFile:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:215) ~[bundleFile:?]
        ... 35 more
2021-11-05 19:00:27.106 [TRACE] [jelectronics.internal.OJCloudHandler] - OJElectronicsCloudHandler.handleRefreshDone(null)
```

@EvilPingu

**Comment by EvilPingu** [comment]
I will have a look.

**Comment by lolodomo** [comment]
Fixed by #11555

**Comment by seime** [comment]
Got a new that put the binding permanently offline;

```
2022-01-25 23:03:43.091 [TRACE] [jelectronics.internal.OJCloudHandler] - OJElectronicsCloudHandler.handleRefreshDone(org.openhab.binding.ojelectronics.internal.models.groups.GroupContentResponseModel@dd6098)
2022-01-25 23:04:14.033 [TRACE] [ics.internal.services.RefreshService] - HTTP-Status 502
2022-01-25 23:04:14.034 [WARN ] [ics.internal.services.RefreshService] - unsupported HTTP-Status 502
2022-01-25 23:04:14.034 [TRACE] [jelectronics.internal.OJCloudHandler] - OJElectronicsCloudHandler.handleConnectionLost()
2022-01-25 23:06:46.904 [TRACE] [jelectronics.internal.OJCloudHandler] - OJElectronicsCloudHandler.handleUnauthorizedWhileSignIn()
```

**Comment by lolodomo** [comment]
Fixed by #12211
